src/__tests__/myOwnSbt.test.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { CheckSbtTokenStatus } from '../components/pages/home/myOwnSbt';
+import '@testing-library/jest-dom';
+import { toast } from 'react-toastify';
+import { wallet } from '../index';
+import {
+  app_contract,
+  gooddollar_contract,
+  near_contract,
+  new_sbt_contract,
+} from '../utils/contract-addresses';
+import { TransferSBT } from '../components/pages/home/MyOwnSbtFiles/transferSbt';
+import { act } from 'react-dom/test-utils';
+
+const accountID = 'megha19.testnet';
+
+// Mock wallet from index.js file
+jest.mock('../index', () => ({
+  wallet: {
+    viewMethod: jest.fn(),
+    callMethod: jest.fn(),
+    accountId: accountID,
+  },
+}));
+
+jest.mock('react-toastify', () => ({
+  toast: {
+    success: jest.fn(),
+    error: jest.fn(),
+    info: jest.fn(),
+  },
+}));
+
+async function fetchSBTTokens() {
+  await waitFor(() => expect(wallet.viewMethod).toHaveBeenCalledTimes(2));
+  await act(async () => {
+    // Expect the first method call with sbt contract
+    expect(wallet.viewMethod).toHaveBeenNthCalledWith(1, {
+      contractId: app_contract,
+      method: 'sbt_supply_by_owner',
+      args: { account: accountID, ctr: new_sbt_contract },
+    });
+    // Expect the second method call with good dollar contract
+    expect(wallet.viewMethod).toHaveBeenNthCalledWith(2, {
+      contractId: app_contract,
+      method: 'sbt_supply_by_owner',
+      args: { account: accountID, ctr: gooddollar_contract },
+    });
+  });
+}
+
+class ResizeObserverMock {
+  observe() {}
+  unobserve() {}
+  disconnect() {}
+}
+global.ResizeObserver = ResizeObserverMock;
+
+describe('Check Sbt Token Status', () => {
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  it('renders component with token supply of 0', async () => {
+    wallet.viewMethod.mockResolvedValueOnce(0);
+    wallet.viewMethod.mockResolvedValueOnce(0);
+    render(<CheckSbtTokenStatus />);
+    await fetchSBTTokens();
+    expect(screen.getByText(/SBT Tokens you own/i)).toBeInTheDocument();
+    expect(screen.getByText(/0/i)).toBeInTheDocument();
+    expect(
+      screen.getByText(/Transfer SBT between my own accounts/i)
+    ).toBeDisabled();
+  });
+
+  it('renders component with token supply greater than 0', async () => {
+    wallet.viewMethod.mockResolvedValueOnce(50);
+    wallet.viewMethod.mockResolvedValueOnce(50);
+    render(<CheckSbtTokenStatus />);
+    await fetchSBTTokens();
+    expect(screen.getByText(/100/i)).toBeInTheDocument();
+    expect(
+      screen.getByText('Transfer SBT between my own accounts')
+    ).not.toBeDisabled();
```

Comment:
why do we event have that functionality? We should have `Soul Transfer` use case, but not `SBT Transfer`. 
cc: @KazanderDad -- could you have a session with @Megha-Dev-19 and @harjaapdhillon16 to specify that for the UI?
Please make sure there is a task for that (not to be solved in this PR, but has to be in the task). This comment should be closed only when the task is there.